mlflow/utils/process.py

```diff
@@ -68,3 +69,29 @@ def _exec_cmd(
     if throw_on_error and prc.returncode != 0:
         raise ShellCommandException.from_completed_process(prc)
     return prc
+
+
+# A global map storing (function, args_tuple) --> (value, pid)
+_per_process_value_cache_map = {}
+
+
+def cache_return_value_per_process(fn):
+    """
+    A decorator which globally cache the return value of the decorated function.
+    But if current process forked out a new child process, in child process,
+    old cache values are invalidated.
+    """
+
+    @functools.wraps(fn)
+    def wrapped_fn(*args):
```

Comment:
Do we need to add kwargs here?
 
```suggestion
    def wrapped_fn(*args, **kwargs):
```

Reply:
I prefer to force the decorated functions calls with only positional arguments., otherwise,
suppose:
```
def f1(a, b):
   return a + b
```
if allow using kwargs, then we might calls with `f1(1, b=3)` or `f1(a=1, b=3)` or `f1(1, 3)`, which generates different args/kwargs, but they should be treated as the same call.

Reply:
I can add a checking to forbid the decorated function called with key-word arguments.